File: Presentations/WG2_11/GlassBR_Code/fromGOOL/cpp/Control.cpp

#include "InputParameters.hpp"
#include "InputFormat.hpp"
#include "DerivedValues.hpp"
#include "InputConstraints.hpp"
#include "Interpolation.hpp"
#include "Calculations.hpp"
#include "OutputFormat.hpp"
#include "ReadTable.hpp"

#include <algorithm>
#include <iostream>
#include <fstream>
#include <iterator>
#include <string>
#include <math.h>
#include <sstream>
#include <vector>

using namespace GlassBR;
using std::string;
using std::vector;
using std::ifstream;
using std::ofstream;

int main(int argc, const char *argv[]) {
    string filename = argv[1];
    InputParameters inparams = InputParameters();
    get_input(filename, inparams);
    derived_params(inparams);
    check_constraints(inparams);
    vector<double> w_array = read_z_array("TSD.txt");
    vector< vector<double> > data_sd = read_x_array("TSD.txt");
    vector< vector<double> > data_q = read_y_array("TSD.txt");
    vector<double> j_array = read_z_array("SDF.txt");
    vector< vector<double> > data_asprat = read_x_array("SDF.txt");
    vector< vector<double> > data_qstar = read_y_array("SDF.txt");
    double q = interpY(data_sd, data_q, w_array, inparams.sd, inparams.wtnt);
    double q_hat = calc_q_hat(q, inparams);
    double j_tol = calc_j_tol(inparams);
    double j = interpZ(data_asprat, data_qstar, j_array, inparams.asprat, q_hat);
    double q_hat_tol = interpY(data_asprat, data_qstar, j_array, inparams.asprat, j_tol);
    double pb = calc_pb(j, inparams);
    double nfl = calc_nfl(q_hat_tol, inparams);
    double lr = calc_lr(nfl, inparams);
    bool is_safe1 = calc_is_safe1(pb, inparams);
    bool is_safe2 = calc_is_safe2(lr, q);
    display_output("outputfile.txt", q, j, q_hat_tol, pb, lr, nfl, is_safe1, is_safe2, inparams);
    std::cout << "Main has been executed and the results have been written to 'outputfile.txt'." << std::endl;
    
    return 0;
}

File: Presentations/WG2_11/GlassBR_Code/fromGOOL/python/Control.py
from __future__ import print_function
import sys
import math
import InputParameters
import InputFormat
import DerivedValues
import InputConstraints
import Interpolation
import Calculations
import OutputFormat
import ReadTable


filename = sys.argv[1]
inparams = InputParameters.InputParameters()
InputFormat.get_input(filename, inparams)
DerivedValues.derived_params(inparams)
InputConstraints.check_constraints(inparams)
w_array = ReadTable.read_z_array("TSD.txt")
data_sd = ReadTable.read_x_array("TSD.txt")
data_q = ReadTable.read_y_array("TSD.txt")
j_array = ReadTable.read_z_array("SDF.txt")
data_asprat = ReadTable.read_x_array("SDF.txt")
data_qstar = ReadTable.read_y_array("SDF.txt")
q = Interpolation.interpY(data_sd, data_q, w_array, inparams.sd, inparams.wtnt)
q_hat = Calculations.calc_q_hat(q, inparams)
j_tol = Calculations.calc_j_tol(inparams)
j = Interpolation.interpZ(data_asprat, data_qstar, j_array, inparams.asprat, q_hat)
q_hat_tol = Interpolation.interpY(data_asprat, data_qstar, j_array, inparams.asprat, j_tol)
pb = Calculations.calc_pb(j, inparams)
nfl = Calculations.calc_nfl(q_hat_tol, inparams)
lr = Calculations.calc_lr(nfl, inparams)
is_safe1 = Calculations.calc_is_safe1(pb, inparams)
is_safe2 = Calculations.calc_is_safe2(lr, q)
OutputFormat.display_output("outputfile.txt", q, j, q_hat_tol, pb, lr, nfl, is_safe1, is_safe2, inparams)
print("Main has been executed and the results have been written to 'outputfile.txt'.");


